third_party/llvm-project/DWARFEmitter.cpp

```diff
@@ -89,6 +89,9 @@ void DWARFYAML::EmitDebugAbbrev(raw_ostream &OS, const DWARFYAML::Data &DI) {
     encodeULEB128(0, OS);
     encodeULEB128(0, OS);
   }
+  // XXX BINARYEN: end the list with a zero. LLVM works with or without this,
+  //               but other decoders may error.
```

Comment:
Nit: once you raise an upstream issue, would be good to reference it by a link here to make sure this is removed once it's resolved.

Reply:
Oh, I haven't noticed this is a patch to the upstream LLVM not custom code. I guess doesn't matter then.

Reply:
This might be worth fixing upstream anyway. IIUC this means that gimli-rs would fail to parse .debub_abbrev generated by LLVM upstream too?

Reply:
Oh sorry, I did file a bug: https://bugs.llvm.org/show_bug.cgi?id=44511 but I forgot to link it here.

This is in the less frequently used YAML path in LLVM, so normal LLVM output should work in gimli-rs.